temoa/data_processing/database_util.py

```
@@ -41,361 +47,284 @@ def __init__(self, database_path, scenario=None):
-            + "and O.period == '"
-            + str(period)
-            + "'"
-        )
+
+        df_columns = columns + ['flow']
+        return pd.DataFrame(self.cur.fetchall(), columns=df_columns)
+
+    def get_emissions_activity_for_period(self, period: int, region: str | None) -> pd.DataFrame:
+        """Retrieves emissions activity data."""
+        if not self.scenario:
+            raise ValueError('A scenario must be set for output-related queries')
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)